### PR TITLE
LRIS and odds and ends

### PR DESCRIPTION
**Pull request by profxj**
Here are the CHANGES:

- Fix oscan sec in read_lris()
- Fix bad return in tracewave.tilts_find_lines()
- Several doc edits
- Fix handling of maskslits
- Fix flexure crashing

### CODE REVIEW

File: pypeit/core/trace_slits.py [hunk 1]
```diff
@@ -1103,6 +1103,7 @@ def edgearr_from_binarr(binarr, binbpx, medrep=0, min_sqm=30.,
     filt = ndimage.sobel(sqmstrace, axis=1, mode=sobel_mode)
     filt *= (1.0 - binbpx)  # Apply to the bad pixel mask
     # siglev
+    #from IPython import embed; embed(header='1106 trace_slits.py')
```
Comment (jhennawi) [inline pypeit/core/trace_slits.py]: Remove this line please.

File: pypeit/scripts/show_2dspec.py [hunk 2]
```diff
@@ -158,8 +158,8 @@ def main(args):
     cut_max = mean + 4.0 * sigma
     chname_skysub='sciimg-det{:s}'.format(sdet)
     # Clear all channels at the beginning
-    viewer, ch = ginga.show_image(image, chname=chname_skysub, waveimg=waveimg,
-                                  bitmask=mask_in, clear=True)
+    #from IPython import embed; embed(header='2dspec A')
```
Comment (jhennawi) [inline pypeit/scripts/show_2dspec.py]: Please remove this line.

File: pypeit/core/wave.py [hunk 3]
```diff
@@ -309,44 +313,92 @@ def flexure_obj(specobjs, maskslits, method, sky_file, mxshft=None):
...
+            new_sky = flexure_interp(sky_wave, specobj, fdict)
+            # Update dict
+            for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
+                flex_dict[key].append(fdict[key])
+            flex_dict['sky_spec'].append(new_sky)
+
     return flex_list
 
 
+def flexure_interp(sky_wave, specobj, fdict):
```
Comment (kbwestfall) [inline pypeit/core/wave.py]: I have some coding practice issues with this function.  Beyond that though, would this be better placed as a method in the `SpecObj` class?

File: pypeit/spectrographs/keck_lris.py [hunk 4]
```diff
@@ -415,7 +415,7 @@ def bpm(self, shape=None, filename=None, det=None, **null_kwargs):
         # Only defined for det=1
         if det == 1:
             msgs.info("Using hard-coded BPM for det=1 on LRISb")
-            self.bpm_img[:, 0:2] = 1
+            self.bpm_img[:, 0:3] = 1
```
Comment (kbwestfall) [inline pypeit/spectrographs/keck_lris.py]: No space and you don't need the 0:
```
self.bpm_img[:,:3] = 1
```

File: pypeit/core/trace_slits.py (hunk 1 above)
Comment (profxj) [reply pypeit/core/trace_slits.py]: removed

File: pypeit/scripts/show_2dspec.py (hunk 2 above)
Comment (profxj) [reply pypeit/scripts/show_2dspec.py]: removed

File: pypeit/core/procimg.py [hunk 5]
```diff
@@ -735,8 +735,12 @@ def replace_columns(img, bad_cols, replace_with='mean'):
         redges = np.concatenate([redges, np.array([bad_cols.size-1])])
     # Loop on em
     for kk, ledge in enumerate(ledges):
-        lval = img[:,ledge-1]
-        rval = img[:,redges[kk]]
+        if ledge == 0:
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/core/procimg.py]: yeah, I like this better too.  Just doesn't
come as naturally.  So keep requesting.

Fixed.

File: pypeit/core/wave.py (hunk 3 above)
Comment (profxj) [reply pypeit/core/wave.py]: Good call.  Have shoved it there.

I don't like this function either.
I simply moved it from the previous...

File: pypeit/spectrographs/keck_lris.py (hunk 4 above)
Comment (profxj) [reply pypeit/spectrographs/keck_lris.py]: edited

**Comment by profxj** [comment]
Test suites running... 

Stand by

**Comment by profxj** [comment]
Unit tests passed (see #682 )

Dev suite passed except one test:

--- PYPEIT DEVELOPMENT SUITE FAILED 1/23 TESTS ---

Directory: /data/Projects/Python/PypeIt-development-suite/REDUX_OUT/Keck_LRIS_red/multi_600_5000_d560
Command line: run_pypeit keck_lris_red_multi_600_5000_d560.pypeit -o
Running pypeit on keck_lris_red_multi_600_5000_d560.pypeit --- FAILED
    ret = ret / rcount
TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'


Which passed yesterday separately.
Am re-running in another window, but I suggest we pass this PR
@jhennawi and @kbwestfall

**Comment by profxj** [comment]
FWIW, that grating ran through fine on its own (again):

[INFO]    :: pypeit.py 308 reduce_all() - Finished calibration group 0
[INFO]    :: pypeit.py 730 print_end_time() - Execution time: 1h 29m 38.04s
[INFO]    :: run_pypeit.py 96 main() - Data reduction complete
[INFO]    :: run_pypeit.py 98 main() - Generating QA HTML
Wrote: QA/MF_B.html
profx> run_pypeit keck_lris_red_multi_600_5000_d560.pypeit -o

**Comment by kbwestfall** [comment]
Does this mean there's some instability somewhere?  I think this setup failed in my `odds_and_ends` branch (merged with this `lris_grating` branch).

I'm fine with merging this to develop, but we should fix it before merging develop into master.  Agreed?

Do you know where its failing on this setup?

**Comment by profxj** [comment]
I can't figure out where it is failing because I can't make it do it
outside of the Dev suite run..

**Comment by kbwestfall** [comment]
Well, I just ran `./pypeit_test lris -s multi_600_5000_d560` and it passed (in my `odds_and_ends` branch), but it failed before. Ugh..  I say let's merge it and see if we can narrow it down before merging to master.  What's your opinion, @jhennawi ?

**Comment by jhennawi** [comment]
Were these tests passed in the "restrict" branch or the current master branch of the dev suite. restricts does not currently run the correct tests.

**Pull request merged**